v2/yarpcfx/yarpcfx.go

```diff
@@ -191,8 +193,31 @@ func NewRouter(p RouterParams) (RouterResult, error) {
 	for _, pl := range p.ProcedureLists {
 		procedures = append(procedures, pl...)
 	}
-	router := yarpcrouter.NewMapRouter("foo" /* Derive from servicefx. */, procedures)
+
+	proceduresWithMW := proceduresWithMiddleware(procedures, p.UnaryTransportMiddleware)
```

Comment:
optional: I think we can reuse the `procedures` variable here and overwrite the previous `[]yarpc.TransportProcedure`.
```
procedures = .applyMiddlewareToProcedures(procedures, p.UnaryTransportMiddleware)
```

We could even write it in-line in the `yarpcrouter.NewMapRouter` constructor if we felt that was sufficiently readable.

```
router := yarpcrouter.NewMapRouter(
  "foo",
  applyMiddlewareToProcedures(procedures),
)
```